File: algorithms/sorting/swift/BubbleSort.swift
import Foundation

public func bubbleSort(_ vetor: [Int]) -> [Int] {
    guard vetor.count > 1 else { return vetor }
    
    var a = vetor
    for i in 0..<a.count-1{
        for j in 0..<a.count-i-1{
            if a[j] > a[j+1] {
                swap(&a[j+1], &a[j])
            }
        }
    }
    return a
}

File: algorithms/sorting/c/bubble-sort.c
//
//  main.c
//  bubbleSort
//
//  Created by Pedro Manhez Naresi on 8/11/17.
//  Copyright © 2017 Pedro Manhez Naresi. All rights reserved.
//

#include <stdio.h>

void printArray(int vetor[], int tamanho){
    for(int i = 0; i < tamanho; i++){
        printf("%d ", vetor[i]);
    }
    printf("\n");
}

void swap(int menor, int i, int vetor[]){
    int aux;
    aux = vetor[menor];
    vetor[menor] = vetor[i];
    vetor[i] = aux;
}

void bubbleSort(int vetor[], int tamanho){
    for(int i = 0; i < tamanho-1; i++){
        for(int j = 0; j < (tamanho - i - 1); j++){
            if(vetor[j] > vetor[j+1]){
                swap((j+1), j, vetor);
                printArray(vetor, tamanho);
            }
        }
    }
    printArray(vetor, tamanho);
}

int main(void) {
    int x;
    scanf("%d", &x);
    int vet[x];
    for(int i = 0; i < x; i++){
        scanf("%d", &vet[i]);
    }
    bubbleSort(vet, x);
}
